tests/unit/computed-test.js

```diff
@@ -0,0 +1,56 @@
+import { get, set } from '@ember/object';
+import autoComputed from 'auto-computed';
+import { module, test } from 'qunit';
+
+module('auto computed properties');
+
+test('passes dependent keys into function as arguments', function(assert) {
+  let obj = {
+    first: 'rob',
+    last: 'jackson',
+
+    @autoComputed('first', 'last')
+    name(first, last) {
+      assert.equal(first, 'rob');
+      assert.equal(last, 'jackson');
+    }
+  };
+
+  get(obj, 'name');
+});
+
+test('dependent key changes invalidate the computed property', function(assert) {
+  let obj = {
+    first: 'rob',
+    last: 'jackson',
+
+    @autoComputed('first', 'last')
+    name(first, last) {
+      return `${first} ${last}`;
+    }
+  };
+
+  assert.equal(get(obj, 'name'), 'rob jackson');
+  set(obj, 'first', 'al');
+  assert.equal(get(obj, 'name'), 'al jackson');
+});
+
+test('works with es6 class', function(assert) {
+  assert.expect(2);
+
+  class Foo {
+    constructor() {
+      this.first = 'rob';
+      this.last = 'jackson';
+    }
+
+    @autoComputed('first', 'last')
+    fullName(first, last) {
+      assert.equal(first, 'rob');
+      assert.equal(last, 'jackson');
+    }
+  }
+
+  let obj = new Foo();
+  get(obj, 'fullName');
+});
```

Comment:
probably can add a few more tests such as `@computed` with no params.

Reply:
I'd also like to see some tests for getters and setters, I think it can be done with:

```
@autoComputed
foo = {
  get() {},
  set() {}
}
```

Reply:
@pzuraq Ah I think I may be missing something :)

So is the purpose of the ember-decorators pkg to provide an `@computed` decorator that works with `getters`?  I had initially thought this `auto-computed` pkg was for functions directly.

Also, did you want a decorator that wraps an object as in the example above?

Reply:
Not native ES getters and setters, but the kind you would provide `computed`, e.g.

```js
Ember.Object.extend({
  foo: computed(function() {

  }),

  bar: computed({
    get() {},
    set() {}
  })
}); 
```

Since all the decorator does is extract the value of the property on the prototype, then replace it with `computed(prop)`, the example I posted above should work as intended already, no changes necessary